aicsimageio/readers/tiff_reader.py

```diff
@@ -0,0 +1,287 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from typing import Dict, Tuple
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from dask import delayed
+from fsspec.spec import AbstractFileSystem
+from tifffile import TiffFile, TiffFileError
+
+from .. import exceptions, types
+from ..dimensions import DimensionNames
+from ..utils import io_utils
+from .reader import Reader
+
+###############################################################################
+
+# "Q" is used by Gohlke to say "unknown dimension"
+# https://github.com/cgohlke/tifffile/blob/master/tifffile/tifffile.py#L10840
+UNKNOWN_DIM_CHAR = "Q"
+
+###############################################################################
+
+
+class TiffReader(Reader):
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str) -> bool:
+        try:
+            with fs.open(path) as open_resource:
+                with TiffFile(open_resource):
+                    return True
+
+        except (TiffFileError, TypeError):
+            return False
+
+    def __init__(self, image: types.PathLike):
+        """
+        Wraps the tifffile API to provide the same aicsimageio Reader API but for
+        volumetric Tiff (and other tifffile supported) images.
+
+        Parameters
+        ----------
+        image: types.PathLike
+            Path to image file to construct Reader for.
+        """
+        # Expand details of provided image
+        self.fs, self.path = io_utils.pathlike_to_fs(image, enforce_exists=True)
+        self.extension = self.path.split(".")[-1]
+
+        # Enforce valid image
+        if not self._is_supported_image(self.fs, self.path):
+            raise exceptions.UnsupportedFileFormatError(self.extension)
+
+    @property
+    def scenes(self) -> Tuple[int]:
+        if self._scenes is None:
+            with self.fs.open(self.path) as open_resource:
```

Comment:
This seems like a common operation, would this make sense to have as an internal method/context manager?
Edit: Even the `TiffFile` part could be included too

Reply:
Yea. I see that but I just don't think its worth it personally. They are simple enough lines. If it was doing something a bit more, I would agree. We may eventually get there if we need to configure `fsspec` interactions / caching or something but for now I am going to leave it.